src/main/java/com/gmail/llmdlio/townyflight/TownyFlightCommand.java

```diff
@@ -185,24 +185,52 @@ private long parseSeconds(String string) {
 	}
 
 	private void parseTownCommand(String[] args) {
-		if (args.length < 2) { 
+		if (args.length < 1) { 
 			showTflyHelp();
 			return;
 		}
 
-		Town town = TownyAPI.getInstance().getTown(args[0]);
-		if (town == null) {
-			Message.of(String.format(Message.getLangString("noTownFound"), args[0])).serious().to(sender);
-			return;
-		}
+		if(args[0].equalsIgnoreCase("toggleflight") && sender instanceof Player) {
+			Resident resident = TownyAPI.getInstance().getResident((Player)sender);
+			if (resident == null) {
+				Message.of(String.format(Message.getLangString("notMayorMsg"))).to(sender);
+				return;
+			}
+
+			if (!resident.isMayor()) {
+				Message.of(String.format(Message.getLangString("notMayorMsg"))).to(sender);
+				return;
+			}

```

Comment:
This can be combined to a `if (resident == null || !resident.isMayor())` block. I would also think this should have its own permission node since I'm sure most servers wouldn't want mayors being able to allow public flight in their towns without the ability to restrict it.

Further, error messages should have the `.serious()` used in order to make it appear as an error.